pennylane/labs/resource_estimation/templates/qubitize.py

```diff
@@ -32,12 +32,25 @@
 class ResourceQubitizeTHC(ResourceOperator):
     r"""Resource class for Qubitization of THC Hamiltonian.
 
+    .. note::
+
+            This decomposition assumes that an appropriately sized phase gradient state is available.
+            Users should ensure that the cost of constructing this state has been accounted for.
+            See also :class:`~.pennylane.labs.resource_estimation.ResourcePhaseGradient`.
+
     Args:
         compact_ham (~pennylane.labs.resource_estimation.CompactHamiltonian): a tensor hypercontracted
             Hamiltonian for which the walk operator is being created
-        coeff_precision (float, optional): precision for loading the coefficients of Hamiltonian
-        rotation_precision (float, optional): precision for loading the rotation angles for basis rotation
-        compare_precision (float, optional): precision for comparing two numbers
+        coeff_precision_bits (int, optional): Number of bits used to represent the precision for loading
+            the coefficients of the Hamiltonian. If :code:`None` is provided, the default value from the
+            :code:`resource_config` is used.
+        rotation_precision_bits (int, optional): Number of bits used to represent the precision for loading
+            the rotation angles for basis rotation. If :code:`None` is provided, the default value from the
+            :code:`resource_config` is used.
+        select_swap_depths (Union[None, int, Iterable(int)],optional): A parameter of :code:`QROM`
+            used to trade-off extra qubits for reduced circuit depth. A list can be used to configure the
+            ``select_swap_depth`` individually for :code:`ResourcePrepTHC` and :code:`ResourceSelectTHC` circuits,
+            respectively.
```

Comment:
Wait! Why are we providing these as inputs! These are all default optional arguments for PrepTHC and SelectTHC. 

Wouldn't it be much nicer if the class' docstring was just: 
`ResourceQubitizeTHC(compact_ham, PrepTHC=None, SelectTHC=None)`

This way if users really want to configure those details they just instantiate the PrepTHC and SelectTHC routines and pass them in. Otherwise, you use the default values and generate PrepTHC and SelectTHC under the hood using the compact_ham.